### PR TITLE
lib: remove `nixvimTypes` alias

### PR DESCRIPTION
**Pull request by MattSturgeon**
Remove the `nixvimTypes` alias. Follow up to #2324.

### CODE REVIEW

File: lib/default.nix [hunk 1]
```diff
@@ -120,13 +120,18 @@ lib.fix (
...
-    maintainers = throw "The `maintainers` alias has been removed. Use `lib.maintainers` on a lib with nixvim's extensions.";
-
-    # TODO: Deprecate the old `nixvimTypes` alias?
-    nixvimTypes = self.extendedLib.types;
-
     toLuaObject = self.lua.toLua;
     mkLuaInline = self.lua.mkInline;
   }
+  //
+    lib.mapAttrs
```
Comment (GaetanLepage) [inline lib/default.nix:127]: Maybe add a TODO to drop this as well in a few months.

**Review by GaetanLepage — COMMENTED** [review]
I checked the rest of the diff and saw no error.

**Comment by MattSturgeon** [comment]
> I checked the rest of the diff and saw no error.

Awesome thanks! I've pushed up a removal TODO as suggested

**Review by traxys — APPROVED** [review]

**Comment by MattSturgeon** [comment]
@mergifyio queue

**Pull request merged**